### PR TITLE
AISDK-90: Add a Custom Vocabulary to the async example

### PR DESCRIPTION
**Pull request by KostasRev**

### CODE REVIEW

File: examples/async_example.py [hunk 1]
```diff
@@ -15,13 +15,24 @@
...
 client = apiclient.RevAiAPIClient(access_token)
 
+# Construct CustomVocabulary objects using our premade class for ease
+mit_professors = CustomVocabulary(
+    ["Robert Berwick", "Noam Chomsky", "Evelina Fedorenko"]
+)
+# Or manually construct CustomVocabulary as such
+other_scientists = {"phrases": ["Albert Einstein"]}
+# Important: place custom vocabularies in a list before passing them into
+# either of client.submit_job_local_file() or sclient.ubmit_job_url()
```
Comment (nishchalb) [inline examples/async_example.py]:
```suggestion
# either of client.submit_job_local_file() or client.submit_job_url()
```

File: examples/async_example.py [hunk 2]
```diff
@@ -15,13 +15,24 @@
...
+custom_vocabularies = [mit_professors, other_scientists]
```
Comment (nishchalb) [inline examples/async_example.py:34]: To me, it's unclear why these are passed in as a list. Is each individual vocabulary treated differently?

Comment (KostasRev) [reply examples/async_example.py:34]: Our [API](https://www.rev.ai/docs#operation/SubmitTranscriptionJob) expects a list of CustomVocabulary instances. And an instance of CustomVocabulary contains a group of phrases. 

A CustomVocabulary instance can therefore contain contextually related custom phrases, and these may then be grouped together into a list in any desired combination to fit a user's needs for a certain transcription job.

File: examples/async_example.py [hunk 3]
```diff
@@ -15,13 +15,24 @@
...
 
+# Construct CustomVocabulary objects using our premade class for ease
+mit_professors = CustomVocabulary(
+    ["Robert Berwick", "Noam Chomsky", "Evelina Fedorenko"]
+)
+# Or manually construct CustomVocabulary as such
+other_scientists = {"phrases": ["Albert Einstein"]}
+# Important: place custom vocabularies in a list before passing them into
+# either of client.submit_job_local_file() or client.submit_job_url()
+custom_vocabularies = [mit_professors, other_scientists]
```
Comment (kbridbur) [inline examples/async_example.py:34]: We should use these in the job submission below

**Review by kbridbur — APPROVED** [review]
One minor comment, looks good otherwise

File: examples/async_example.py (hunk 2 above)
Comment (amikofalvy) [reply examples/async_example.py:34]: @nishchalb we made this design decision early on with the idea that we could support combining prebuilt custom-vocabularies in the future. At this point though, this does not appear to be a feature that customers want. This API interface gives us more flexibility in the future.

**Pull request merged**